lib/ci-ecr-stack.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  Construct, NestedStack, RemovalPolicy, Stack, StackProps,
+} from '@aws-cdk/core';
+import { CfnPublicRepository } from '@aws-cdk/aws-ecr';
+import {
+  ArnPrincipal, IRole, ManagedPolicy, PolicyStatement, Role,
+} from '@aws-cdk/aws-iam';
+
+export interface EcrStackProps extends StackProps {
+  /** Should we deploy ECR */
+  readonly deployECR?: boolean;
+  /** Main Node role arn */
+  readonly mainNodeAccountNumber: string;
+  /** removal policy for the ECR Repositories */
+  readonly removalPolicy?: RemovalPolicy;
+  /** should we skip creating ecr  */
+  readonly createEcrRepositories?: boolean;
+}
+
+export class CiEcrStack extends NestedStack {
+  constructor(scope: Construct, id: string, props: EcrStackProps) {
+    super(scope, id, props);
+    if (props.createEcrRepositories ?? false) {
+      CiEcrStack.createRepositories(scope, this, props.removalPolicy ?? RemovalPolicy.RETAIN);
+    }
+    const ecrPolicy = CiEcrStack.createEcrPolicy(this, id);
+
+    CiEcrStack.createEcrRole(this, ecrPolicy, props.mainNodeAccountNumber);
+  }
+
+  public static createEcrRole(stack: Stack, ecrPolicy: ManagedPolicy, mainNodeAccountNumber: string) : IRole {
+    return new Role(stack, 'ecr-stack-role', {
+      roleName: 'OpenSearch-CI-ECR-ecr-role',
```

Comment:
Once ECR is enough to say its an ECR role, rename to `OpenSearch-CI-ecr-role`

Reply:
`OpenSearch-CI-ECR` is the stack name and ecr-role is the role name

Reply:
Why not grab the stack name from `stack.name` property then?

Reply:
`stack.name` would also contain the env where the stack is being deployed. Which would cause the role name to change according to the env, creating a problem when using roles in `jenkinsfile`